Add .gitignore content in advanced section

rwnfoo opened the pull request:
#240 

_Note_: Branched off of `pat-content` hence why it includes PAT stuff

## How to view files:
```
git checkout rwnfoo/gitignore
docsify serve docs
```

beardofedu commented on docs/23_gitignore.md:
```diff
@@ -0,0 +1,44 @@
+## Configure Git to ignore files
+
+You can add a text file in your repository titled `.gitignore` that instructs Git to stop tracking changes of certain files. This also means that these files will not be pushed to GitHub from your local machine.
```
```suggestion
While working within your repository, you might notice files that your preferred editor or operating system have place in the repository. These files are typically things that you don't want git to track or include in the history of your project. Although you can manually ignore those files by never committing them to the history of the project, there is an easier way for you to ignore those files. Git enables you to ignore specific files, files types, and folders through the use of a text file you can add to your  repository titled `.gitignore`. This file instructs Git to stop tracking changes of certain files, which means that these files will not be pushed to GitHub from your local machine.
```

beardofedu commented on docs/23_gitignore.md:
```diff
@@ -0,0 +1,44 @@
+
+[This website](https://www.toptal.com/developers/gitignore) is another resource for you to find appropriate `.gitignore` templates when your project contains multiple programming languages.
+
+![gitignore.io](img/gitignore.io.png)
```
I don't know if we need this image necessarily, while the other images provide context regarding the use of GitHub provided `.gitignore` solutions, we don't have much insight as to any upcoming changes to this site so we might not be aware of any UI changes coming down the pipe. Including / Excluding this image from the file isn't a blocker, just my $.02

beardofedu commented on docs/23_gitignore.md:
````diff
@@ -0,0 +1,44 @@
+
+In your Terminal or Git Bash, `cd` to the repository of interest and create a `.gitignore` file
+```
+touch .gitignore
````
In general can we try to be more consistent when presenting the reader with steps to complete a task? In this specific instance, the user has created a `.gitignore` file, but hasn't committed it, added anything to the `.gitignore` file, etc. 

Ideally we would have some exercise that lets us show them how the `.gitignore` file works both from a "I can see how I could potentially use this" perspective and a "Oh, I can see it in action through this activity" perspective. 

Overall, great addition for this topic though 🎉

parkerbxyz replied:
This might be useful here: https://gist.github.com/octocat/9257657

parkerbxyz commented on docs/23_gitignore.md:
```diff
@@ -0,0 +1,44 @@
+## Configure Git to ignore files
+
+While working within your repository, you might notice files that your preferred editor or operating system have place in the repository. These files are typically things that you don't want git to track or include in the history of your project. Although you can manually ignore those files by never committing them to the history of the project, there is an easier way for you to ignore those file ...
```
```suggestion
While working within your repository, you might notice system or configuration files created by your operating system or text editor. These are typically files that you don't want Git to track or include in the history of your project. You can configure Git to ignore specific files, files types, and folders by defining them in a text file titled `.gitignore` in your repository's root directory. 
```

parkerbxyz commented on docs/23_gitignore.md:
````diff
@@ -0,0 +1,44 @@
+```
+touch .gitignore
+```
+
````
```suggestion

To share the ignore rules with other users who clone the repository, commit the `.gitignore` file in to your repository.

```

parkerbxyz commented on docs/23_gitignore.md:
```diff
@@ -0,0 +1,44 @@
+
+### Here are some ways you can add a `.gitignore` file:
+
+#### Ignore files  locally
```
```suggestion
#### Ignore files for a single repository
```

parkerbxyz commented on docs/23_gitignore.md:
```diff
@@ -0,0 +1,44 @@
+
+While working within your repository, you might notice files that your preferred editor or operating system have place in the repository. These files are typically things that you don't want git to track or include in the history of your project. Although you can manually ignore those files by never committing them to the history of the project, there is an easier way for you to ignore those file ...
+
+**Files that you might want Git to ignore:**
+1. Secrets, e.g. `*.pem`, `*.key`
+1. Binary files, e.g. `*.pdf*`, `*.docx`
+1. Log files, e.g. `*.log`
+1. Build files, e.g. `Build/*`
```
Maybe this section should be moved down below the steps to create an empty `.gitignore` file.

parkerbxyz commented on docs/23_gitignore.md:
```diff
@@ -0,0 +1,46 @@
+**Files that you might want Git to ignore:**
+
+1. Secrets, e.g. `*.pem`, `*.key`
+1. Binary files, e.g. `*.pdf*`, `*.docx`
```
```suggestion
1. Binary files, e.g. `*.pdf`, `*.docx`
```

rwnfoo commented:
@beardofedu @parkerbxyz Thank you for the feedback! I incorporated all of them. Changes I made: 

- Add activity to commit a `.gitignore` file, using [this template](https://gist.github.com/octocat/9257657) 
- Moved `github/gitignore` and `gitignore.io` content up such that the activities are at the bottom of the page, consistent with formats in other pages
- Removed `gitignore.io` image

beardofedu commented on docs/23_gitignore.md:
````diff
@@ -2,47 +2,110 @@
+1. In your Terminal or Git Bash, `cd` to a repository of interest and create a `.gitignore` file:
 
 ```sh
 touch .gitignore
````
`touch` doesn't work if the user is not using a bash-based terminal. Windows CMD users can use: `echo >> .gitignore` to accomplish the same thing.

beardofedu commented on conflict-template/HEAD:
```diff
@@ -0,0 +1 @@
+ref: refs/heads/master
```
Where did all these `conflict-template/*` files come from? I think they were added by mistake....

rwnfoo replied:
@beardofedu oh shoot, you're right, another reason why `git add . ` is baaad 😞
